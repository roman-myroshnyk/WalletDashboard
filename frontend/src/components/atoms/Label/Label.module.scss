@import '@/styles/colors.scss';

.inputLink {
  color: $darkGray;
  font-weight: bold;
  font-size: 20px;
  font-family: Epilogue, sans-serif;
  font-style: normal;
  line-height: 24px;
  text-decoration: none;
}
