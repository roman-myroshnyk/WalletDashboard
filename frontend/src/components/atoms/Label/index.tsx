import { FC, ReactNode } from 'react';
import styles from './Label.module.scss';

export interface IProps {
    htmlFor?: string;
    children: ReactNode;
}

const Label:FC<IProps> = ({ htmlFor, children }) => (
  <label className={styles.inputLink} htmlFor={htmlFor}>
    {children}
  </label>
);

Label.defaultProps = {
  htmlFor: 'id',
};
export default Label;
