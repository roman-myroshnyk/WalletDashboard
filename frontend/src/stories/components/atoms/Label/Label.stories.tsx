import { Meta, Story } from '@storybook/react/types-6-0';
import Label, { IProps } from '@/atoms/Label';

export default {
  title: 'atoms/Label',
  component: Label,
  argTypes: {
    text: {
      controls: 'text',
    },
  },
} as Meta;

interface ITemplateProps extends IProps {
  text: string;
}

const Template:Story<ITemplateProps> = ({ ...args }) => (
  <Label {...args}>
    {args.text}
  </Label>
);

export const Label4USD2ETH = Template.bind({});

Label4USD2ETH.args = {
  text: 'USD/ETH',
  htmlFor: 'id1',
};

export const Label4EUR2ETH = Template.bind({});

Label4EUR2ETH.args = {
  text: 'EUR/ETH',
  htmlFor: 'id2',
};
