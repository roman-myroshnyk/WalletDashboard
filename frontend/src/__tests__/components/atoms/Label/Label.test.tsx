import '.../../../mocks/framer-motion';
import { render } from '@testing-library/react';
import Label from '@/atoms/Label';

describe('atoms/Label', () => {
  it('Label should render', () => {
    const { container } = render(
      <Label htmlFor="myInput">
        My label
      </Label>,
    );
    expect(container).toMatchSnapshot();
  });
});
