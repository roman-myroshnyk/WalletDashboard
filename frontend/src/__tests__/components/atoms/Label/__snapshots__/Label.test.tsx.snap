// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Label Label should render 1`] = `
<div>
  <label
    class="inputLink"
    for="myInput"
  >
    My label
  </label>
</div>
`;
